cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    /**
+     * Converts the values from the Iterable into String values and adds the pairs to the current set.
+     *
+     * @param iterable Key/Value pairs
+     * @return this instance for chaining
+     * @throws InvalidConfigParameterException if value is null or is of an unsupported type
+     */
+    public OrderedProperties addIterablePairs(Iterable<? extends Map.Entry<String, ? extends Object>> iterable) {
+        for (Map.Entry<String, ? extends Object> entry : iterable) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+
+            if (value instanceof String) {
+                pairs.put(key, (String) value);
+            } else if (value instanceof Integer || value instanceof Long || value instanceof Boolean
+                || value instanceof Double || value instanceof Float) {
+                pairs.put(key, String.valueOf(value));
+            } else if (value == null) {
+                throw new InvalidConfigParameterException(key, "A null value is not allowed for this key");
+            } else {
+                throw new InvalidConfigParameterException(key,
+                    " - Unsupported type " + value.getClass() + " in configuration for this key");
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Parse key/value pairs and add to the current pair set.
+     * N.B. values containing a newline or other control characters will probably cause problems!
+     * @param keyValuePairs Pairs in key=value format, pairs separated by newlines
+     * @throws InvalidConfigParameterException if value is null
+     * @return this instance for chaining
+     */
+    public OrderedProperties addStringPairs(String keyValuePairs) {
+        try (LineNumberReader lnr = new LineNumberReader(new StringReader(keyValuePairs))) {
+            for (; ; ) {
+                String kv = lnr.readLine();
+                if (kv == null) {
+                    break;
+                }
+                int equals = kv.indexOf('=');
+                if (equals < 0) {
+                    throw new InvalidConfigParameterException(kv, "A null value is not allowed for this key");
+                } else {
+                    pairs.put(kv.substring(0, equals), kv.substring(equals + 1));
+                }
+            }
+        } catch (IOException e) {
+            // A StringReader should not be able to throw an IOException
+            throw new IllegalStateException(e);
+        }
+        return this;
+    }
+
+    /**
+     * Add a key/value pair
+     * @param key The key
+     * @param value The value
+     * @return this instance for chaining
+     */
+    public OrderedProperties addPair(String key, String value) {
+        pairs.put(key, value);
+        return this;
+    }
+
+    /**
+     * Generate a string with pairs formatted as key=value separated by newlines.
+     * N.B. values containing a newline or other control characters will probably cause problems!
```

Comment:
This concerns me. We previously used `Properties` which is an old and somewhat crufty API, but has very well-defined behaviour wrt to things like `\n` in values. I also think that we rely on the serialized format being compatible with `java.util.Properties` because the configuration eventually gets passed to Kafka brokers etc as a `.properties` file. I think the contract for this method should be that it produces `.properties` compatible output.

Reply:
Are there any other points where properties compatibility is important?

Reply:
Updated class and added multiple new unit tests to ensure compatibility with Properties format